File: NS/2017/F3/codigo.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int encontra_cadeia(char **S, int ind);

int main () {
    int N;
    char **cadeia;
    int i, achou = -1;

    scanf("%d", &N);

    cadeia = (char *) malloc(N * sizeof(char));
    for (i = 0; i < N; i++)
        cadeia[i] = (char *) malloc(11 * sizeof(char));

    for (i = 0; i < N; i++) 
        scanf(" %[^\n]s", cadeia[i]);

    for (i = 1; i < N && achou == -1; i++)
        if(encontra_cadeia(cadeia, i))
            achou = i;

    if (achou == -1)
        printf("ok");
    else
        printf("%s", cadeia[achou]);

    return 0;
}

int encontra_cadeia(char **S, int ind) {
    int i, j;
    char copy[22];

    for (i = 0; i < ind; i++) 
    {   for (j = 0; j < ind; j++)
        {   if (i != j) {
                strcpy(copy, S[i]);
                strcat(copy, S[j]);

                if (strstr(copy, S[ind]) != NULL)
                    return ind;
            }
        }
    }

    return -1;
}

File: NS/2017/F3/codigo.py
def encontraSubstring (s, n):
    for i in range(n):
        for j in range(n):
            if i != j:
                if s[n] in s[i]+s[j]:
                    return n

    return -1

S = []
N = int(input())

for i in range(N):
    S.append(input())

i = 0
achou = False
while i < N and achou == False:
    if encontraSubstring(S, i) != -1:
        ind = encontraSubstring(S, i)
        achou = True
    
    i += 1

if achou == True:
    print(S[ind])
else:
    print("ok")